### PR TITLE
feat: shadowgraph wiring

### LINKED ISSUE #10935: Move shadowgraph into wiring framework
The shadowgraph is currently maintained on the consensus scheduler. It should have its own separate scheduler.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #10935 

[Wiring diagram](https://mermaid.ink/svg/Zmxvd2NoYXJ0IExSCnYwKCgiQCIpKQpzdWJncmFwaCB2MVsiQ29uc2Vuc3VzIFBpcGVsaW5lIl0Kc3ViZ3JhcGggdjJbIkxpbmtlZCBFdmVudCBJbnRha2UiXQp2M3t7ImdldE5vbkFuY2llbnRFdmVudFdpbmRvdyJ9fQp2NFsibGlua2VkRXZlbnRJbnRha2U8YnIgLz7imIUiXQp2NXt7ImxpbmtlZEV2ZW50SW50YWtlU3BsaXR0ZXIifX0KZW5kCnY2WyJpbk9yZGVyTGlua2VyPGJyIC8+ypgiXQplbmQKc3ViZ3JhcGggdjdbIkludGFrZSBQaXBlbGluZSJdCnN1YmdyYXBoIHY4WyJFdmVudCBWYWxpZGF0aW9uIl0KdjlbImV2ZW50RGVkdXBsaWNhdG9yPGJyIC8+ypgiXQp2MTBbImV2ZW50U2lnbmF0dXJlVmFsaWRhdG9yPGJyIC8+ypgiXQp2MTFbImludGVybmFsRXZlbnRWYWxpZGF0b3I8YnIgLz7igKAiXQplbmQKc3ViZ3JhcGggdjEyWyJPcnBoYW4gQnVmZmVyIl0KdjEzWyJvcnBoYW5CdWZmZXI8YnIgLz7KmCJdCnYxNHt7Im9ycGhhbkJ1ZmZlclNwbGl0dGVyIn19CmVuZAp2MTVbLyJldmVudEhhc2hlciIvXQplbmQKc3ViZ3JhcGggdjE2WyJQQ0VTIl0KdjE3Wy8iZXZlbnREdXJhYmlsaXR5TmV4dXMiL10KdjE4Wy8icGNlc1NlcXVlbmNlciIvXQp2MTlbInBjZXNXcml0ZXI8YnIgLz5Ac8qYIl0KZW5kCnN1YmdyYXBoIHYyMFsiU3RhdGUgRmlsZSBNYW5hZ2VtZW50Il0KdjIxe3siZXh0cmFjdE9sZGVzdE1pbmltdW1HZW5lcmF0aW9uT25EaXNrIn19CnYyMnt7InNhdmVUb0Rpc2tGaWx0ZXIifX0KdjIzWyJzaWduZWRTdGF0ZUZpbGVNYW5hZ2VyIl0KdjI0e3sidG9TdGF0ZVdyaXR0ZW5Ub0Rpc2tBY3Rpb24ifX0KZW5kCnN1YmdyYXBoIHYyNVsiU3RhdGUgU2lnbmF0dXJlIENvbGxlY3Rpb24iXQp2MjZ7eyJhbGxTdGF0ZXNSZXNlcnZlciJ9fQp2Mjd7eyJjb21wbGV0ZVN0YXRlRmlsdGVyIn19CnYyOHt7ImNvbXBsZXRlU3RhdGVzUmVzZXJ2ZXIifX0KdjI5e3siZXh0cmFjdENvbnNlbnN1c1NpZ25hdHVyZVRyYW5zYWN0aW9ucyJ9fQp2MzB7eyJleHRyYWN0UHJlY29uc2Vuc3VzU2lnbmF0dXJlVHJhbnNhY3Rpb25zIn19CnYzMXt7InJlc2VydmVkU3RhdGVTcGxpdHRlciJ9fQp2MzJbInN0YXRlU2lnbmF0dXJlQ29sbGVjdG9yIl0KZW5kCnYzM1tbImFwcGxpY2F0aW9uVHJhbnNhY3Rpb25QcmVoYW5kbGVyIl1dCnYzNFsiZXZlbnRDcmVhdGlvbk1hbmFnZXI8YnIgLz7KmOKZoSJdCnYzNVsiaGVhcnRiZWF0Il0KdjM2Wy8icGNlc1JlcGxheWVyIi9dCnYzNygoInMiKSkKdjM4WyJzaGFkb3dncmFwaCJdCnYzOVsic3RhdGVTaWduZXIiXQp2NDAoKCLKmCIpKQp2NDEoKCLigKAiKSkKdjQyKCgi4piFIikpCnY0MygoIuKZoSIpKQp2MjYgLS0gInN0YXRlcyIgLS0+IHYyNwp2MjYgLS0gInN0YXRlcyIgLS0+IHYyMgp2MzMgLS4gImZ1dHVyZXMiIC4tbyB2NDIKdjI3IC0tPiB2MjgKdjM0IC0uICJub24tdmFsaWRhdGVkIGV2ZW50cyIgLi0+IHY0MQp2OSAtLSAiZXZlbnRzIHdpdGggdW52YWxpZGF0ZWQgc2lnbmF0dXJlcyIgLS0+IHYxMAp2MTUgLS0gIm5vbi12YWxpZGF0ZWQgZXZlbnRzIiAtLT4gdjExCnYxMCAtLSAidW5vcmRlcmVkIGV2ZW50cyIgLS0+IHYxMwp2MjkgLS0gImNvbnNlbnN1cyBzaWduYXR1cmUgdHJhbnNhY3Rpb25zIiAtLT4gdjMyCnYyMSAtLSAibWluaW11bSBpZGVudGlmaWVyIHRvIHN0b3JlIiAtLT4gdjM3CnYzMCAtLSAicHJlY29uc2Vuc3VzIHNpZ25hdHVyZSB0cmFuc2FjdGlvbnMiIC0tPiB2MzIKdjMgLS4gIm5vbi1hbmNpZW50IGV2ZW50IHdpbmRvdyIgLi0+IHY0MAp2MzUgLS0gImhlYXJ0YmVhdCIgLS0+IHY0Mwp2NiAtLSAibGlua2VkIGV2ZW50cyIgLS0+IHY0CnY2IC0tICJsaW5rZWQgZXZlbnRzIiAtLT4gdjM4CnYxMSAtLSAibm9uLWRlZHVwbGljYXRlZCBldmVudHMiIC0tPiB2OQp2NCAtLSAicm91bmQgbGlzdHMiIC0tPiB2NQp2NCAtLSAiZmx1c2ggcmVxdWVzdCIgLS0+IHYxOQp2NSAtLSAicm91bmRzIiAtLT4gdjMKdjEzIC0tICJldmVudCBsaXN0cyIgLS0+IHYxNAp2MTQgLS0gInByZWNvbnNlbnN1cyBldmVudHMiIC0tPiB2MzMKdjE0IC0tICJwb3NzaWJsZSBwYXJlbnRzIiAtLT4gdjM0CnYxNCAtLSAicHJlY29uc2Vuc3VzIGV2ZW50cyIgLS0+IHYzMAp2MTQgLS0gImV2ZW50cyB0byBzZXF1ZW5jZSIgLS0+IHYxOAp2MzYgLS0gImRvbmUgc3RyZWFtaW5nIHBjZXMiIC0tPiB2MAp2MzYgLS0gImV2ZW50cyB0byBoYXNoIiAtLT4gdjE1CnYxOCAtLSAidW5saW5rZWQgZXZlbnRzIiAtLT4gdjYKdjE4IC0tICJldmVudHMgdG8gd3JpdGUiIC0tPiB2MTkKdjE5IC0tICJsYXRlc3QgZHVyYWJsZSBzZXF1ZW5jZSBudW1iZXIiIC0tPiB2MTcKdjMxIC0tPiB2MjYKdjIyIC0tICJzdGF0ZXMgdG8gc2F2ZSIgLS0+IHYyMwp2MjMgLS0gInN0YXRlIHNhdmluZyByZXN1bHQiIC0tPiB2MjEKdjIzIC0tICJzdGF0ZSBzYXZpbmcgcmVzdWx0IiAtLT4gdjI0CnYzMiAtLSAicmVzZXJ2ZWQgc3RhdGUgbGlzdHMiIC0tPiB2MzEKY2xhc3NEZWYgczAgZmlsbDojZjg4LHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjAsdjM3LHY0MCx2NDEsdjQyLHY0MyBzMApjbGFzc0RlZiBzMSBmaWxsOiM5Q0Ysc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOjJweApjbGFzcyB2MSx2MTYsdjIwLHYyNSx2NyBzMQpjbGFzc0RlZiBzMiBmaWxsOiNCRUYsc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOjJweApjbGFzcyB2MTIsdjIsdjggczIKY2xhc3NEZWYgczMgZmlsbDojY2NjLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjE0LHYxNSx2MTcsdjE4LHYyMSx2MjIsdjI0LHYyNix2MjcsdjI4LHYyOSx2Myx2MzAsdjMxLHYzNix2NSBzMwpjbGFzc0RlZiBzNCBmaWxsOiNmZjksc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOjJweApjbGFzcyB2MTAsdjExLHYxMyx2MTksdjIzLHYzMix2MzMsdjM0LHYzNSx2MzgsdjM5LHY0LHY2LHY5IHM0Cg==?bgColor=e8e8e8)

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java [hunk 1]
```diff
@@ -188,10 +204,11 @@ private void wire() {
         pcesWriterWiring
                 .latestDurableSequenceNumberOutput()
                 .solderTo(eventDurabilityNexusWiring.latestDurableSequenceNumber());
-
         signedStateFileManagerWiring
                 .oldestMinimumGenerationOnDiskOutputWire()
                 .solderTo(pcesWriterWiring.minimumAncientIdentifierToStoreInputWire());
+
+        linkedEventIntakeWiring.nonExpiredEventWindowOutput().solderTo(shadowgraphWiring.nonExpiredEventWindowInput());
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:211]: I'm curious why this wasn't wired to receive the NonAncientEventWindow like everything else?

**Review by edward-swirldslabs — DISMISSED** [review]

**Comment by poulok** [comment]
I really need a legend to understand the wiring diagrams now. It has gotten large, and I don't know which boxes are which type, or what the funky characters at the bottom mean, or how to interpret the red circles.

**Review by hendrikebbers — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowGraphSynchronizer.java [hunk 2]
```diff
@@ -174,10 +174,7 @@ public ShadowGraphSynchronizer(
         this.nonAncestorFilterThreshold = syncConfig.nonAncestorFilterThreshold();
 
         this.filterLikelyDuplicates = syncConfig.filterLikelyDuplicates();
-        this.latestEventTipsetTracker = latestEventTipsetTracker;
-        if (filterLikelyDuplicates) {
-            Objects.requireNonNull(latestEventTipsetTracker);
-        }
+        this.latestEventTipsetTracker = Objects.requireNonNull(latestEventTipsetTracker);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowGraphSynchronizer.java:177]: `latestEventTipsetTracker` should now be annotated `@NonNull`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/LinkedEventIntakeWiring.java [hunk 3]
```diff
@@ -36,13 +36,15 @@
  * @param consensusRoundOutput        the output wire for consensus rounds
  * @param nonAncientEventWindowOutput the output wire for the {@link NonAncientEventWindow}. This output is transformed
  *                                    from the consensus round output
+ * @param nonExpiredEventWindowOutput the output wire for the minimum generation non-ancient.
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/LinkedEventIntakeWiring.java:39]: Incorrect doc

**Review by alittley — APPROVED** [review]

**Comment by cody-littley** [comment]
Note: before merging this, must first wait for the PR that adds the expired threshold to the non-ancient event window.

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowGraphSynchronizer.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowGraphSynchronizer.java]: the `latestEventTipsetTracker` data structure no longer exists

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/LinkedEventIntakeWiring.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/LinkedEventIntakeWiring.java]: this wire no longer exists

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java]: Until very recently, non-ancient event window didn't have the expired event threshold. Now that the event window does have the expiration threshold, this is wired up like everything else.

**Review by jeromy-cannon — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/ShadowgraphWiring.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Wiring for the {@link com.swirlds.platform.gossip.shadowgraph.ShadowGraph}.
+ *
+ * @param eventInput                 the input wire for events to be added to the shadow graph
+ * @param nonExpiredEventWindowInput the input wire for the non-expired event window
+ */
+public record ShadowgraphWiring(
+        @NonNull InputWire<EventImpl> eventInput,
+        @NonNull InputWire<NonAncientEventWindow> nonExpiredEventWindowInput,
+        @NonNull Runnable flushRunnable) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/ShadowgraphWiring.java:36]: nit: missing doc for `flushRunnable`

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/com/swirlds/platform/test/consensus/ShadowGraphEventObserver.java [hunk 5]
```diff
@@ -51,7 +53,7 @@ public ShadowGraphEventObserver(@NonNull final ShadowGraph shadowGraph) {
      */
     @Override
     public void consensusRound(final ConsensusRound consensusRound) {
-        shadowGraph.expireBelow(consensusRound.getGenerations().getMinRoundGeneration());
+        shadowGraph.updateNonExpiredEventWindow(consensusRound.getNonAncientEventWindow());
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/com/swirlds/platform/test/consensus/ShadowGraphEventObserver.java:56]: This class is no longer used outside of tests

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/module-info.java [hunk 6]
```diff
@@ -6,8 +6,10 @@
     requires transitive com.swirlds.platform.core;
     requires com.swirlds.config.api.test.fixtures;
     requires com.swirlds.config.api;
+    requires com.swirlds.logging;
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/module-info.java:9]: Were the modifications in this file intentional?

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncNode.java [hunk 7]
```diff
@@ -255,7 +256,7 @@ public ShadowGraphSynchronizer getSynchronizer() throws InterruptedException {
     }
 
     /**
-     * <p>Calls the {@link ShadowGraph#expireBelow(long)} method and saves the {@code expireBelow} value for use in
+     * <p>Calls the {@link ShadowGraph#updateNonExpiredEventWindow(com.swirlds.platform.consensus.NonAncientEventWindow)} method and saves the {@code expireBelow} value for use in
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncNode.java:259]: Nit: line length

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java [hunk 8]
```diff
@@ -589,10 +591,20 @@ private void runFallenBehindTest(
         executor.setCustomPreSyncConfiguration((c, l) -> {
             when(c.getConsensus().getMinGenerationNonAncient()).thenReturn(callerMinGen);
             when(c.getConsensus().getMaxRoundGeneration()).thenReturn(callerMaxGen);
-            c.getShadowGraph().expireBelow(callerMinGen);
+            c.getShadowGraph()
+                    .updateNonExpiredEventWindow(new NonAncientEventWindow(
+                            0 /* ignored by shadowgraph */,
+                            0 /* ignored by shadowgraph */,
+                            callerMaxGen,
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java:598]: The original test has `callerMinGen`, but you changed it to `callerMaxGen`. Was that intentional?

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/ShadowgraphWiring.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/ShadowgraphWiring.java:36]: added

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/com/swirlds/platform/test/consensus/ShadowGraphEventObserver.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/com/swirlds/platform/test/consensus/ShadowGraphEventObserver.java:56]: Yup, I'm aware of that. That's why I moved this into the test directory. It's currently used by `TestIntake`, which is why I did not remove it entirely.

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/module-info.java (hunk 6 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/main/java/module-info.java:9]: Yes, unfortunately. The CI job that tells you how to change the `module-info.java` files required me to make this change.

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncNode.java (hunk 7 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncNode.java:259]: Auto-formatted this file. Annoying how spotless doesn't format javadocs.

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java (hunk 8 above)
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java:598]: Unintentional. Fixed.

**Review by alittley — DISMISSED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Pull request merged**